qiskit/mapper/_layout.py

```diff
@@ -69,6 +75,19 @@ def __delitem__(self, key):
         dict.__delitem__(self, self[key])
         dict.__delitem__(self, key)
 
+    @staticmethod
+    def checktype(thing):
```

Comment:
why does this need to be a staticmethod? a private method of self seems more appropriate.

Reply:
Static methods also warrantee that they will not have side effects in `self`. `checktype` is pure with respect to self. I can rename it as `_checktype` if that preferable.